jujugui/static/gui/src/app/jujulib/index.js

```diff
@@ -25,97 +20,59 @@ var module = module;
 (function (exports) {
 
   /**
-     Utility function for making requests via the bakery.
+    Serializes an object into a query string.
 
-     _makeRequest
-     @param bakery {Object} The bakery object to use.
-     @param path {String} The JEM endpoint to make the request from,
-         e.g. '/model'
-     @param method {String} The type of http method to use, e.g. GET or POST.
-     @param params {Object} Optional data object to sent with e.g. POST
-        commands.
-     @param callback {Function} A callback to handle errors or accept the data
-         from the request. Must accept an error message or null as its first
-         parameter and the response data as its second.
-     @param parse {Boolean} Whether or not to parse the response as JSON.
-     @param redirect {Boolean} Whether or not to redirect on a 401 within
-        the bakery.
+    @param obj {Object} the object to serialize
+    @return a query string serialized from the object.
   */
-  const _makeRequest = function(
-    bakery, url, method, params, callback, parse, redirect) {
-    const wrappedCallback = (err, response) => {
+  const serializeObject = function(obj) {
+    return Object.keys(obj).map(p =>
+        `${encodeURIComponent(p)}=${encodeURIComponent(obj[p])}`).join('&');
+  };
+  /**
+    @param {Function} callback The callback to wrap.
+    @param {Object} options Any options for the callback wrapper.
+      parseJSON: {Boolean} false - Whether the response should be passed
+        through JSON.parse.
+    @return {Function} The wrapped callback.
+  */
+  const _wrap = (callback, options={}) => {
+    return (err, response) => {
       if (err) {
         callback(err, null);
         return;
       }
-      const text = response.target.responseText;
+      const target = response.target;
+      const text = target && target.responseText;
       if (!text) {
         callback(null, null);
         return;
       }
-      // TODO frankban: we should really avoid defaulting optional
-      // parameters to true.
-      if (parse !== undefined && !parse) {
+      if (!options.parseJSON) {
         callback(null, text);
         return;
       }
-      let jsonResponse;
+      let jResp;
       try {
-        jsonResponse = JSON.parse(text);
+        jResp = JSON.parse(text);
+        err = jResp.error || jResp.Error || jResp.message || jResp.Message;
```

Comment:
Why are we considering a response with a message as an error? In pre-existing code we only lookup Error and error.